kratos/mpi/utilities/parallel_fill_communicator.cpp

```diff
@@ -413,7 +418,7 @@ void ParallelFillCommunicator::InitializeParallelCommunicationMeshes(
     // Call the sub model part.
     for (ModelPart& r_sub_model_part : rModelPart.SubModelParts())
     {
-        ComputeCommunicationPlan(r_sub_model_part);
+        ComputeCommunicationPlan(r_sub_model_part, rModelPart.GetCommunicator().GetDataCommunicator());
```

Comment:
I feel that having some parts with the model part data communicator and execute with other data communicator is ambiguous and a source of obscure errors...

Reply:
all of them have the same DataComm if you check the flow of the code
I can also pass it down directly but then I have to change a signature
in the end is the same

Reply:
improved :)